### PR TITLE
(PR58) Documentation/Bump Antares Simulator Version

### LINKED ISSUE #145: Antares Simulator v10.1.0 detected - validation required
## New Version Detected

A new version of Antares Simulator has been detected.

| | |
|---|---|
| **Previous version** | `10.0.1` |
| **New version** | `10.1.0` |
| **Release page** | https://github.com/AntaresSimulatorTeam/Antares_Simulator/releases/tag/v10.1.0 |

This issue tracks validation and integration into GEMS.

---

## What's Changed
* chore(deps): bump SonarSource/sonarqube-scan-action from 7.0.0 to 7.1.0 by @dependabot[bot] in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3544
* Remove ortools dependency from antares-problem-generator [ANT-4846] by @payetvin in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3538
* Add behave tests for simple tests [ANT-4643] by @flomnes in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3545
* replace raw Area* with unique_ptr in AreaList by @YoshuaLisasi in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3550
* chore(version): 10.0.1 by @JasonMarechal25 in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3511
* Fix: CentOS7 link and segfault by @JasonMarechal25 in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3561
* Fix remix hydro sts levels computation [ANT-4888] by @Juliette-Gerbaux in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3557
* ci: Use fixed submodule for Antares_Simulator_Tests_NR instead of branch main by @flomnes in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3568
* refacto: remove duplication in "sorties" by @JasonMarechal25 in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3541
* refacto: Clean up scenario-builder by @flomnes in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3567
* Fix(doc): modeler doc by @flomnes in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3529
* feat: Multiple Hydro Rule Curves by @nikolaredstork in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2829

**Full Changelog**: https://github.com/AntaresSimulatorTeam/Antares_Simulator/compare/v10.0.1...v10.1.0

---

## Triage

> Review the release notes and open the relevant process issue(s):
> - Impact on GEMS Language docs → open a **DOC-01** issue
> - Impact on model libraries or taxonomies → open a **LT-01** issue

- [ ] Release notes reviewed
- [ ] DOC-01 issue opened (if GEMS Language docs are affected)
- [ ] LT-01 issue opened (if libraries or taxonomies are affected)
- [ ] No action needed (if release has no GEMS impact)

---

### Validation
- [ ] Run E2E tests on library models
- [ ] Run E2E tests on taxonomy
- [ ] Run edge case scenarios
- [ ] Compare results with previous version

### Compatibility
- [ ] Check GEMS Language syntax impact
- [ ] Verify Modeler compatibility
- [ ] Identify breaking changes

### Documentation
- [ ] Update GEMS Language documentation if needed
- [ ] Document behavioral changes
- [ ] Add migration notes

### Decision
- [ ] Approve version - update `dependencies.json` `antares_simulator_version` to `10.1.0`
- [ ] Reject version - close this issue with explanation

---

## Notes

_Add findings and observations here._

### PR DESCRIPTION
**Pull request by GuillaumeMaistre**
## Process ID

**Process:**
DOC-01

## Description

Based on #145 
Bump documentation based on the antares simulator version

## Impact Analysis

YES, as we change the antares-simulator version

## Checklist

- [x] E2E tests pass (or confirmed not affected)
- [x] Library version bumped if library changed (`library.version` in `libraries/<library>.yml`)
- [x] Changelog entry added if library changed (`libraries/CHANGELOG-<library>.md`)
- [x] Release notes entry added if language-level change (`doc/0_Home/4_release_notes.md`)
- [x] Compatibility matrix updated if applicable (`COMPATIBILITY.md`)
- [x] Documentation updated or confirmed up to date
- [x] `AGENTS.md` reviewed for impact and updated if needed

## Triage (from issue #145)

- [x] Release notes reviewed
- [x] DOC-01 issue opened (if GEMS Language docs are affected)
- [x] LT-01 issue opened (if libraries or taxonomies are affected)
- [ ] No action needed (if release has no GEMS impact)

### CODE REVIEW

File: COMPATIBILITY.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ This table maps documentation/language versions to the interpreter and library v
 | v0.3.0        | 10.0.1            | —      |       |
 | v0.3.1        | 10.0.1            | —      |       |
 | v0.3.2        | 10.0.1            | —      |       |
+| v0.3.3        | 10.0.1            | —      |       |
```
Comment (aoustry) [inline COMPATIBILITY.md:10]: Maybe add also v.0.3.4 | 10.1.0 ?

**Review by aoustry — APPROVED** [review]
Approved but please answer the comment

File: COMPATIBILITY.md (hunk 1 above)
Comment (GuillaumeMaistre) [reply COMPATIBILITY.md:10]: I added this version
